### PR TITLE
Remove warning about empty packages pre 3.0

### PR DESCRIPTION
**Pull request by jeremiedimino**
The testsuite is broken following #4867. Also, I didn't noticed but some tests ended being updated in a way that they didn't test what they were originally testing.

I tried to update tests but it was a pain so I removed the warning pre 3.0 and also set `allow_empty` to `true` for packages defined by an opam file rather than a package stanza. This makes the behaviour more conservative.

However, we still need a cram test for #4867.

### CODE REVIEW

File: src/dune_engine/package.ml [hunk 1]
```diff
@@ -644,6 +644,12 @@ let decode ~dir =
      and+ allow_empty =
        field_b "allow_empty"
          ~check:(Dune_lang.Syntax.since Stanza.syntax (3, 0))
+     and+ lang_version = Dune_lang.Syntax.get_exn Stanza.syntax in
+     let allow_empty =
+       if lang_version < (3, 0) then
```
Comment (rgrinberg) [inline src/dune_engine/package.ml:649]: `lang_version < (3, 0) || allow_empty`

**Comment by jeremiedimino** [comment]
Most of it is a revert of the changes to the testsuite in #4861. Merging as the testsuite is currently broken.

**Review by rgrinberg — APPROVED** [review]
Thanks.

File: src/dune_engine/package.ml (hunk 1 above)
Comment (jeremiedimino) [reply src/dune_engine/package.ml:649]: Looks better, I pushed it to main

**Pull request merged**